trunk/src/hpinterp/update_gdof.F90

```diff
@@ -437,3 +440,534 @@ subroutine update_gdof()
+!
+!    routine name       - update_gdof_par
+!
+!-----------------------------------------------------------------------
+!> @brief OpenMP parallel version of update_gdof
+!> @date Mar 2023
+!-----------------------------------------------------------------------
+!
+subroutine update_gdof_par
+!
+   use constrained_nodes
+   use data_structure3D
+   use element_data
+   use environment, only: QUIET_MODE
+   use MPI        , only: MPI_COMM_WORLD,MPI_INTEGER,MPI_REAL8,MPI_SUM, &
+                          MPI_MIN,MPI_IN_PLACE,MPI_STATUS_IGNORE
+   use mpi_param  , only: RANK,ROOT,NUM_PROCS
+   use par_mesh   , only: DISTRIBUTED,HOST_MESH
+   use GMP
+!
+   implicit none
+!
+   integer :: ntype
+   character(len=6) :: mdltype
+!
+!..orientation of element nodes
+   integer, dimension(12) :: nedge_orient
+   integer, dimension(6)  :: nface_orient
+!
+!..element nodes
+   integer, dimension(27)      :: nodesl, norientl
+   integer, dimension(MAXNODM) :: nodm
+!
+!..order of approximation for an element
+   integer, dimension(19) :: norder
+!
+!..reference coordinates for an element
+   real(8), dimension(3,8) :: xsub
+!
+!..geometry dofs for an element
+   real(8), dimension(3,MAXbrickH) :: xnod
+!
+!..temporary for OpenMP parallelism
+   real(8) :: coord(3,MAXquadH)
+!
+!..auxiliary variables for timing
+   real(8) :: MPI_Wtime,start_time,end_time
+!
+!..auxiliary variables
+   integer :: iel,iv,ie,ifc,ind,iflag,i,k,loc
+   integer :: mdle,no,nod,nr_elem_nodes,nrnodm,nr_up_elem
+!
+!..additional variables for distributed case
+   integer :: src,rcv,tag,count,ierr,j_loc,j_glb,j_off,loc_max
+   integer :: ndofH,ndofE,ndofV,ndofQ
+   logical :: nod_flg
+   integer :: nod_cnt(NUM_PROCS)
+   integer, allocatable :: nod_loc(:),nod_tmp(:),nod_glb(:),nod_rnk(:)
+   real(8), dimension(:,:), pointer :: buf
+!
+!..OpenMP parallelization
+   integer, allocatable :: nodes_elem(:,:), nodes_irreg(:,:), nrnodes_irreg(:)
+   integer :: nodesi(27)
+   integer :: j, sz, nrnodi
+!
+!-----------------------------------------------------------------------
+!
+   call MPI_BARRIER(MPI_COMM_WORLD, ierr)
+   start_time = MPI_Wtime()
+!
+!$OMP PARALLEL DO
+!..lower the GMP interface flag for all nodes
+   do nod=1,NRNODS
+      NODES(nod)%geom_interf=0
+   enddo
+!$OMP END PARALLEL DO
+!
+!-----------------------------------------------------------------------
+!
+!..fetch active elements
+   if (.not. DISTRIBUTED) then
+      ELEM_SUBD(1:NRELES) = ELEM_ORDER(1:NRELES)
+      NRELES_SUBD = NRELES
+   endif
+!
+   allocate(nodes_elem(27,NRELES_SUBD))
+   allocate(nodes_irreg(27,NRELES_SUBD))
+   allocate(nrnodes_irreg(NRELES_SUBD))
+!
+!-----------------------------------------------------------------------
+!
+! Get nodes for each element in subd
+!
+!$omp parallel default(shared)
+!$omp do private(iel,mdle,nodesl,norientl,nodm,nrnodm,   &
+!$omp            nr_elem_nodes,loc,nod,i,j,ntype)        &
+!$omp    schedule(static) reduction(+:nr_up_elem)
```

Comment:
I think it's good for a code to be stylistically consistent, and OMP pragmas are everywhere capitalized throughout the hp3D library, so I think we should follow that convention.

Reply:
This is actually one of my pet peeves; I don't get why we capitalize all of the OMP calls. I can capitalize the '!$OMP' but `parallel, schedule, etc.` has no reason to be capitalized, it just makes things less readable IMO.

Reply:
In this case, I think we'd made the choice to capitalize deliberately because many text editors show fortran omp sections by default just as commented sections. By capitalizing, they're more distinguishable, emphasizing they are not just comments. In the end, it's just a stylistic choice but either way I think it should be consistent throughout the library code.